database.ts

```diff
@@ -247,6 +260,16 @@ export class Datebase_View implements ProfileController, EventGetter, EventRemov
         })();
         return res;
     }
+
+    private async recordRelay(eventID: string, url: string) {
+        const records = this.relayRecords.get(eventID);
+        if (records) {
+            records.add(url);
+        } else {
+            this.relayRecords.set(eventID, new Set([url]));
+        }
+        await this.relayRecorder.setRelayRecord(eventID, url);
```

Comment:
Both `this` and `relayRecorder` implement the same interface. We can declare the same interface in the class definition. Not a big deal.